test.sh

```diff
@@ -510,19 +510,28 @@ function test_make_run_tests() {
 # Smoke Tests: Run a small collection of fast-running unit-tests
 # ##################################################################
 function run_fast_unit_tests() {
+
    "$BINDIR"/unit/splinterdb_quick_test
    "$BINDIR"/unit/btree_test
    "$BINDIR"/unit/util_test
+
    "$BINDIR"/unit/misc_test
    "$BINDIR"/unit/limitations_test
+
+   # Default execution runs w/ no background threads configured
    "$BINDIR"/unit/task_system_test
+
+   # Just exercise with some combination of background threads to ensure
+   # that basic usage of background thread config params still works.
+   "$BINDIR"/unit/task_system_test --num-normal-bg-threads 4 --num-memtable-bg-threads  2
```

Comment:
I strongly prefer that "unit" tests have sensible default options, rather than requiring command line flags, so that someone can just run the executable by name and get useful coverage & feedback.
Passing flags should be the exceptional case, not the one used by the test script.

I feel less strongly about this for larger system tests.

Reply:
@rosenhouse - The 'problem' is that this test creates one task-system right up at the top in the `SETUP()` method.

We want to test that task-system creation can be done in both ways: with and w/o bg-threads configured.
I felt that adding these invocations in `test.sh` w/ desired config options to exercise this functionality is not a bad thing. You will anyway get the coverage when you run quick unit-tests.

To do what you ask I will probably have to add a new test sub-case to tear down the task system and recreate it anew w/ diff config params.

If you insist on this point, it will probably mean a rework of this `task_system_test`. 

Let me know what you want based on this response, and I'll rework accordingly, if needed.

Reply:
Here's my belief: if the case is worth covering, then it should be covered when running the test without flags.  But I'm curious what others thing too.

Reply:
Rework the unit test case to refactor the `task_system_create()` call into a sub-fn that is called from all cases.

Reply:
Done, in 2 newly added sub-test-cases.